### PR TITLE
docs: add Using JSON Web Key Sets section

### PR DESCRIPTION
**Pull request by hassankhan**
**Description of Changes Made (if issue reference is not provided)**

Rewrite introduction to Authentication & Authorization, and add new section with information about JWT/JWK integration.

### CODE REVIEW

**Comment by keydunov** [comment]
Good job 👍 It'd worth adding also how we cache and re-fetch the keys in case of the rotating JWKs.

File: docs/content/Authentication-&-Authorization.md [hunk 1]
```diff
@@ -62,19 +57,105 @@ Authentication is handled outside of Cube.js. A typical use case would be:
...
-| can still use it to [pass a security context][link-security-context].
+| can still use it to [pass a security context][ref-sec-ctx].
+<!-- prettier-ignore-end -->
+
+## Using JSON Web Key Sets (JWKS)
+
+### Configuration
+
+As mentioned previously, Cube.js supports verifying JWTs using industry-standard
+JWKS. The JWKS can be provided either as a JSON object conforming to [JWK
```
Comment (rpaik) [inline docs/content/Authentication-&-Authorization.md:78]:
```suggestion
JWKS. The JWKS can be provided as a JSON object conforming to [JWK
```

File: docs/content/Authentication-&-Authorization.md [hunk 2]
```diff
@@ -62,19 +57,76 @@ Authentication is handled outside of Cube.js. A typical use case would be:
...
 
 <!-- prettier-ignore-start -->
 [[info |]]
 | **In development mode, the token is not required for authorization**, but you
-| can still use it to [pass a security context][link-security-context].
+| can still use it to [pass a security context][ref-sec-ctx].
 <!-- prettier-ignore-end -->
 
-[link-security-context]: /security#security-context
+## Using JSON Web Key Sets (JWKS)
```
[the remark this answers is not in the document]
Comment (hassankhan) [reply docs/content/Authentication-&-Authorization.md:73]: @keydunov Yep I was planning to either update this PR once https://github.com/cube-js/cube.js/pull/2143 is merged, or add the guides as a followup PR once this is merged.

File: docs/content/Authentication-&-Authorization.md (hunk 1 above)
Comment (hassankhan) [reply docs/content/Authentication-&-Authorization.md]: @rpaik I forgot to mention both ways of configuring JWKS here.

Comment (rpaik) [reply docs/content/Authentication-&-Authorization.md]: Cool, thanks 😄

**Pull request merged**